Modules/@babylonjs/react-native/shared/BabylonNative.cpp

```diff
@@ -0,0 +1,230 @@
+        //       happens during a dev mode reload). In this case, EngineHook.ts won't call NativeEngine.dispose,
+        //       so we need to manually do it here to properly clean up these resources.
+        void Deinitialize()
+        {
+            if (m_disposeEngine)
+            {
+                m_disposeEngine();
+                m_disposeEngine = {};
+            }
+        }
+
+        void UpdateView(void* windowPtr, size_t width, size_t height)
+        {
+            // TODO: We shouldn't have to dispatch to the JS thread most of this, but not doing so results in a crash.
+            //       I don't understand the issue yet, but for now just retain the pre-refactor logic. We'll need to
+            //       resolve this to enable manual non-JS thread rendering. Note this only repros in release builds
+            //       where we actually call ResetView.
+            m_jsCallInvoker->invokeAsync([this, windowPtr, width, height]()
+            {
+                if (!m_graphics)
+                {
+                    m_graphics = Graphics::CreateGraphics(windowPtr, width, height);
+                    m_graphics->AddToJavaScript(m_env);
+                    Plugins::NativeEngine::Initialize(m_env, true);
+                    m_resolveInitPromise();
+                }
+                else
+                {
+                    m_graphics->UpdateWindow(windowPtr);
+                    m_graphics->UpdateSize(width, height);
+                    m_graphics->EnableRendering();
+                }
+            });
+        }
+
+        void ResetView()
+        {
+            // TODO: We shouldn't have to dispatch to the JS thread for this since we are already on the JS thread,
+            //       but there is an issue in NativeEngine where it will Dispatch a call to RenderCurrentFrame, then
+            //       get disposed, then try to actually render the frame. This results in immediately re-enabling
+            //       graphics after disabling it here. For now, retain the pre-refactor logic (queueing on the JS thread).
+            m_jsCallInvoker->invokeAsync([this]()
+            {
+                if (m_graphics)
+                {
+                    m_graphics->DisableRendering();
+                }
+            });
+        }
+
+        void SetPointerButtonState(uint32_t pointerId, uint32_t buttonId, bool isDown, uint32_t x, uint32_t y)
+        {
+            if (isDown)
+            {
+                m_nativeInput->PointerDown(pointerId, buttonId, x, y);
+            }
+            else
+            {
+                m_nativeInput->PointerUp(pointerId, buttonId, x, y);
+            }
+        }
+
+        void SetPointerPosition(uint32_t pointerId, uint32_t x, uint32_t y)
+        {
+            m_nativeInput->PointerMove(pointerId, x, y);
+        }
+
+        jsi::Value get(jsi::Runtime& runtime, const jsi::PropNameID& prop) override
+        {
+            if (prop.utf8(runtime) == "initializationPromise")
```

Comment:
Probably should store `prop.utf8(runtime)` in a variable or it will reallocate the resulting string multiple times.

Reply:
This is a very coarse grained operation. In most apps, it will only happen once. Does caching really matter in this case?

Reply:
Nope, but it's still not a great pattern. I think we should fix it so that no one copies it.

Reply:
Oh, I totally misunderstood what you were originally suggesting. Will fix.